### ISSUE TITLE
Multiple version of tensorflow that exist in system site packages and venv site packages

### ISSUE THREAD

**Issue by kafji** [tensorflow/tensorflow#42978]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: n/a
- TensorFlow installed from (source or binary): binary
- TensorFlow version: nightly (2.4.0.dev20200904)
- Python version: 3.8.2
- Installed using virtualenv? pip? conda?: virtualenv+pip
- Bazel version (if compiling from source): n/a
- GCC/Compiler version (if compiling from source): n/a
- CUDA/cuDNN version: n/a
- GPU model and memory: n/a

**Describe the problem**
Error when importing tensorflow nightly installed in venv while having another version of tensorflow installed system-wide.

**Provide the exact sequence of commands / steps that you executed before running into the problem**
Following steps in https://www.tensorflow.org/install/pip.
```
pip3 install --upgrade tensorflow-cpu
python3 -m venv --system-site-packages venv
source venv/bin/activate
(venv) pip install --upgrade pip
(venv) pip install --upgrade tf-nightly-cpu
(venv) python -c "import tensorflow as tf;print(tf.reduce_sum(tf.random.normal([1000, 1000])))"
```

**Any other info / logs**
Traceback:
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "somewhere/venv/lib/python3.8/site-packages/tensorflow/__init__.py", line 433, in <module>
    _ll.load_library(_main_dir)
  File "somewhere/venv/lib/python3.8/site-packages/tensorflow/python/framework/load_library.py", line 154, in load_library
    py_tf.TF_LoadLibrary(lib)
tensorflow.python.framework.errors_impl.NotFoundError: /home/user/.local/lib/python3.8/site-packages/tensorflow/core/kernels/libtfkernel_sobol_op.so: undefined symbol: _ZN10tensorflow8OpKernel11TraceStringEPNS_15OpKernelContextEb
```

I have tensorflow 2.3.0 installed in system site packages and that causes https://github.com/tensorflow/tensorflow/blob/47aab0b49ef7432261f60430f5e29bb331cd5233/tensorflow/api_template.__init__.py#L138-L142 to load tensorflow kernel from system site packages.

Is there a way to have multiple version of tensorflow that exist in system site packages and venv site packages?

**Comment by bhack** [comment]
Can you try to create the venv without `--system-site-packages`?

**Comment by kafji** [comment]
> Can you try to create the venv without `--system-site-packages`?

Same result. Unless I also comment out https://github.com/tensorflow/tensorflow/blob/47aab0b49ef7432261f60430f5e29bb331cd5233/tensorflow/api_template.__init__.py#L119

Thanks for pointing out that flag. Now I have a workaround.

**Comment by Saduf2019** [comment]
@kafji 
If the issue is resolved please feel free to move this to closed status.

**Comment by bhack** [comment]
@kafji Probably it was a side effect of https://github.com/tensorflow/tensorflow/pull/38663 /cc @angerson @yongtang

**Comment by kafji** [comment]
@bhack I think it was from an older commit https://github.com/tensorflow/tensorflow/commit/5195204b47d1cf9516ff5eea8232fbff8d320521#diff-62ec18ff8bdd93adaff55160f27a7e09R69

**Comment by kafji** [comment]
I'm closing this.

The workaround is to comment out https://github.com/tensorflow/tensorflow/blob/47aab0b49ef7432261f60430f5e29bb331cd5233/tensorflow/api_template.__init__.py#L119 which located in tensorflow's `__init__.py`

**Comment by SiegeLordEx** [comment]
If you don't mind, I'd rather keep this open. It seems quite reasonable to me for TF to load shared libraries from the virtualenv installation rather than the user directory. The current behavior is quite unintuitive.

**Comment by foxik** [comment]
I am also hit by the problem -- with TF 2.4.0rc3, `pip install` in a virtual environment followed by `import tensorflow as tf` fails with
```
~/.local/lib/python-3.6/site-packages/tensorflow/core/kernels/libtfkernel_sobol_op.so: undefined symbol: _ZN10tensorflow8OpKernel11TraceStringEPNS_15OpKernelContextEb
```

This makes virtual environments unusable with TF :-(

**Comment by foxik** [comment]
BTW, shouldn't the `site.USER_SITE` be used only when `site.ENABLE_USER_SITE` is True?

**Comment by foxik** [comment]
I created a PR #45399 to address the issue.

**Comment by himaniyadav** [comment]
Which __init__.py should you edit? 
I know to comment  `_site_packages_dirs += [] if _site.USER_SITE is None else [_site.USER_SITE]` this out but in which version of tensorflow, the one in the venv?

**Comment by foxik** [comment]
The issue is fixed in TF 2.4.1 and tf-nightly, so it should be enough to use a new TF.

But if you want to fix it manually, you need to fix the one you are actually importing (which is the one in venv, I suppose).
